src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -0,0 +1,663 @@
+                   |each read, to assign each read to a sample.  Reads that to not match any sample within the given error tolerance
+                   |will be placed in the 'unmatched' file.
+                   |
+                   |The output directory will contain one BAM file per sample in the sample sheet, plus a BAM for reads that could
+                   |not be assigned to a sample given the criteria.  The output files will be the concatenation of sample id, sample
+                   |name, and sample barcode bases (expected not observed), delimited by ".".  A metrics file will also be output
+                   |for sample barcodes.  More information about these metrics can be found here:
+                   |  https://broadinstitute.github.io/picard/picard-metric-definitions.html#SampleBarcodeMetric
+                   |
+                   |The read group's sample id, sample name, and library id all correspond to the similarly named values in the sample sheet.
+                   |Library id will be the sample id if not found, and the platform unit will be the sample name concatenated with the sample
+                   |barcode bases delimited by a ".".
+                   |
+                   |The sample section of the sample sheet should contain information related to each sample with the following keys and values:
+                   |  - Sample Identifier:  Sample_ID
+                   |  - Sample Name:        Sample_Name
+                   |  - Library Identifier: Library_ID
+                   |  - Sample Project:     Sample_Project
+                   |  - Description:        Description
+                   |
+                   |The following are optional values may included in the sample sheet to specify sample barcode bases in any read:
+                   |  - Read One Inline Sample Barcode Bases*: R1_Barcode_Bases
+                   |  - Read Two Inline Sample Barcode Bases*: R2_Barcode_Bases
+                   |  - i7 Sample Barcode Bases:              Index
+                   |  - i5 Sample Barcode Bases:              Index2
+                   |* Note that these are custom extensions and not standard.
+                   |
+                   |The read structures will be used to extract the observed sample barcode and molecular identifiers from each
+                   |read.  The observed sample barcode will be matched to the sample barcodes extracted from the bases in the sample sheet
+                   |and associated read structures.  Please see the following link for the read structure:
+                   |  https://broadinstitute.github.io/picard/javadoc/picard/picard/illumina/parser/ReadStructure.html
+                   |
+                   |""",
+  group = ClpGroups.Utilities)
+class DemuxFastqs
+(
+  @arg(flag="f1", doc="Input fastq file (optionally gzipped) for the first read of paired end data.")
```

Comment:
Input fastq file(s) (optionally gzipped) for the first template read of paired end data.